gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,903 @@
+    "    d['norm'] = sc.reciprocal(variances)\n",
+    "\n",
+    "    d.masks['zero_error'] = sc.equal(variances, 0.0 * variances.unit)\n",
+    "    \n",
+    "    d = sc.sum(d, dim_to_sum)\n",
+    "    sum = d['data']/d['norm']\n",
+    "    return sum\n",
+    "\n",
+    "# If we are not interested in strain regions for the unloaded sample, we can just combine all spectra to improve statistics.\n",
+    "# maybe another masking before summation, with tougher threshold to exclude boderline pixels.\n",
+    "sample_final = weighted_sum(rebinned_wavelength[\"sample\"], \"spectrum\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "rebinned_wavelength.coords[\"detector_mapping\"] = imaging.make_detector_groups(bank_width, bank_width,\n",
+    "                                                                              nx_target, ny_target)\n",
+    "\n",
+    "grouped = sc.groupby(rebinned_wavelength, \"detector_mapping\").sum(\"spectrum\")\n",
+    "grouped_sample_elastic = grouped[\"sample_elastic\"]"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "histogram_matrices = [sample_final, grouped_sample_elastic]\n",
+    "template_histogram = histogram_matrices[1]\n",
+    "\n",
+    "# Average out detector positions\n",
+    "positions_dataset = sc.Dataset()\n",
+    "positions_dataset[\"position\"] = sc.Variable(rebinned_wavelength.coords[\"position\"])\n",
+    "positions_dataset.coords[\"detector_mapping\"] = rebinned_wavelength.coords[\"detector_mapping\"]\n",
+    "grouped_positions = sc.groupby(positions_dataset, \"detector_mapping\").mean(\"spectrum\")\n",
+    "\n",
+    "plots = []\n",
+    "for spec_num in template_histogram.coords[\"detector_mapping\"].values:\n",
+    "    pos = grouped_positions[\"position\"][\"detector_mapping\", int(spec_num)]\n",
+    "    # Append spectrum index and position\n",
+    "    plots.append((spec_num,pos))\n",
+    "\n",
+    "# We cant access position using (dim, I), since the vector 3d has not dim currently\n",
+    "Y_key = 1\n",
+    "X_key = 0\n",
+    "\n",
+    "# We must use round to prevent floating points errors during sorting like -0.05 > 0.05\n",
+    "plots.sort(key=lambda t: (round(-t[1].value[Y_key], 10), round(t[1].value[X_key], 10)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Bragg edge position roughly, in Angstrom\n",
+    "FCC_a = 3.5 # Aangstrom, taken from COD entry 9008469\n",
+    "BCC_a = 2.865 # Angstrom, average value from COD\n",
+    "indices_FCC = [(1,1,1),(2,0,0),(2,2,0),(3,1,1)]#,(2,2,2)]#,(4,0,0)]\n",
```

Comment:
I suspect that Peter has commented out the reflections that are not visible because they would fall out of the wavelength range. i.e. the bragg edges will not appear in the data. Minor point: Would be nice to verify this by calculation and made a comment to this effect. Maybe could just be clarified with Peter.

Reply:
I've added a comment to remind us to ask

Reply:
Let's just ask him now as he's on slack. I'll open discussion.